src/Application/Migrations/15_function_ticket_state.sql

```diff
@@ -129,4 +129,88 @@ END
 $$
 LANGUAGE plpgsql;
 
+-- function also resets handle_id and failed flag
+CREATE OR REPLACE FUNCTION ticket_reset(param_ticket_id bigint, param_ticket_types enum_ticket_type[] DEFAULT array[]::enum_ticket_type[])
+  RETURNS TABLE (ticket_id bigint,  from_state enum_ticket_state, to_state enum_ticket_state, follows_meta_ticket boolean, follows_encoding_ticket boolean) AS
+$$
+DECLARE
+  ticket tbl_ticket%rowtype;
+  inital_state enum_ticket_state;
+BEGIN
+  --first get the full given ticket
+  SELECT * INTO ticket FROM tbl_ticket where id = param_ticket_id AND (array_length(param_ticket_types, 1) IS NULL OR (array_position(param_ticket_types, ticket_type) IS NOT NULL));
```

Comment:
I'm not sure, if it useful or not to return at this point.
That means also if children or depending tickets matches the filter, they will not be reset if the parent ticket doesn't match.

Reply:
I mean, missing an actual use-case I don't have a strong opinion on this.
I could see both cases (returning or not) being useful.
But not being exposed by an API it is even less useful ;)

Reply:
So add a parameter to the function to control the behavior?

Reply:
Mh, no. Leave it as is, add the parameters to the RPC API and model and let us merge this PR.
When we feel we need to control this behavior further, we can change the code again, no?